mlflow/tensorflow.py

```diff
@@ -633,28 +636,51 @@ def autolog(every_n_iter=100):
 
     @gorilla.patch(tensorflow.estimator.Estimator)
     def export_saved_model(self, *args, **kwargs):
+        global _AUTO_END_RUN
```

Comment:
Hm maybe we should patch `estimator.train` instead? Users may not necessarily export their Estimator as a SavedModel after fitting it

Reply:
I see what you mean in terms of those workflows, but one issue with patching `estimator.train` is that we currently don't have a way to log a `tf.estimator` model after training. 

- If we were to log a model after training, we would require a `serving_input_receiver_fn`, which would then require knowledge of the specific input to the model - how would we do this? If we were to do this, we could just remove the patch over `export_saved_model` since `estimator.train` would do that for us.
- We could potentially save the models as checkpoints after training? Though checkpoints are more heavyweight than `export_saved_model` and would require a reworking of `load_model` to support loading checkpoints (which would be a decent amount of work...).
- If we were to end the run after training and leave the model logging to `export_saved_model`, then each model would be logged in a separate run from its parameters and metrics - not very ideal.

Additionally, I have `_AUTO_END_RUN` as a global since `_log_event` is usually invoked before the patched functions for TensorBoard logs, leading to cases where a run is created by `autolog` in `_log_event` and not properly ended after fitting a model. 

Let me know if there's something I missed and if there's an elegant solution to this issue.

Reply:
After offline discussion, we decided that the value of ending an automatically-started run is greater than the value of ending the run after the user calls `export_saved_model`. The first option lends itself well to workflows, and users can manually manage their runs if they want to save their models to their corresponding runs.

I'll close this PR and open two, one for Keras-specific autolog ending, and one for `tf.estimator` autolog ending.